### PR TITLE
[Azure Site Recovery] PR contains: aligning PowerShell to service object model, supporting some new features and some bug fixes.

### PR DESCRIPTION
**Pull request by avneeshrai**
[Azure Site Recovery] Commit contains:
1) Aligning PowerShell to service object model.
2) Introducing deprecation warning for cmdlets and parameter sets to be deprecated in future.
2) HUB changes.
3) Bug Fixes.

### CODE REVIEW

**Comment by cormacpayne** [comment]
@avneeshrai Hey Avneesh, please merge the latest changes to dev to fix the conflicts

**Comment by avneeshrai** [comment]
@cormacpayne I have merged the latest changes from dev and resolved the conflicts. Please review and merge the PR.

**Comment by avneeshrai** [comment]
@markcowl Could you please review this PR.

File: src/ResourceManager/SiteRecovery/Commands.SiteRecovery/Network/RemoveAzureRMSiteRecoveryNetworkMapping.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ namespace Microsoft.Azure.Commands.SiteRecovery
     /// </summary>
     [Cmdlet(VerbsCommon.Remove, "AzureRmSiteRecoveryNetworkMapping")]
     [OutputType(typeof(ASRJob))]
-    public class RemoveAzureRMSiteRecoveryNetworkMapping : SiteRecoveryCmdletBase
```
Comment (shahabhijeet) [inline src/ResourceManager/SiteRecovery/Commands.SiteRecovery/Network/RemoveAzureRMSiteRecoveryNetworkMapping.cs]: >RemoveAzureRMSiteRecoveryNetworkMapping [](start = 17, length = 39)

Any reason the remove Networking mapping does not have SupportShouldProcess parameter?

Comment (avneeshrai) [inline src/ResourceManager/SiteRecovery/Commands.SiteRecovery/Network/RemoveAzureRMSiteRecoveryNetworkMapping.cs]: We get errors from the build infra for SupportShouldProcess parameter only if cmdlet has -Force parameter.  I guess if Force is not present SupportShouldProcess is not mandatory as PR build succeeds.

---
In reply to: [78791145](https://github.com/Azure/azure-powershell/pull/2912#discussion_r78791145) [](ancestors = 78791145)

File: src/ResourceManager/SiteRecovery/Commands.SiteRecovery.Test/ScenarioTests/SiteRecoveryTests.ps1 [hunk 2]
```diff
@@ -359,4 +359,47 @@ function Test-SiteRecoveryVaultCRUDTests
...
-}
\ No newline at end of file
+}
+
+
+<#
+.SYNOPSIS
+Site Recovery Fabric Tests New model
+#>
+function Test-SiteRecoveryFabricTest
```
Comment (shahabhijeet) [inline src/ResourceManager/SiteRecovery/Commands.SiteRecovery.Test/ScenarioTests/SiteRecoveryTests.ps1]: >SiteRecoveryFabricTest [](start = 14, length = 22)

Is this good enough test coverage for anyone to contribute to your feature as this is a public repo? I think we need more scenario tests or at least bare minimum unit tests.

**Comment by shahabhijeet** [comment]
Overall looks good. Let's get the tests in to get decent coverage of your feature.

**Comment by avneeshrai** [comment]
Sure..i will update the PR with needed tests.

---
In reply to: [247099534](https://github.com/Azure/azure-powershell/pull/2912#issuecomment-247099534) [](ancestors = 247099534)

File: src/ResourceManager/SiteRecovery/Commands.SiteRecovery.Test/ScenarioTests/SiteRecoveryTests.ps1 (hunk 2 above)
Comment (avneeshrai) [inline src/ResourceManager/SiteRecovery/Commands.SiteRecovery.Test/ScenarioTests/SiteRecoveryTests.ps1]: Sure..i will update the PR with needed tests.

---
In reply to: [78799081](https://github.com/Azure/azure-powershell/pull/2912#discussion_r78799081) [](ancestors = 78799081)

**Comment by avneeshrai** [comment]
Updated the PR after taking your comments and including new tests.

---
In reply to: [247124518](https://github.com/Azure/azure-powershell/pull/2912#issuecomment-247124518) [](ancestors = 247124518,247099534)

File: src/ResourceManager/SiteRecovery/Commands.SiteRecovery.Test/ScenarioTests/SiteRecoveryTests.ps1 (hunk 2 above)
Comment (avneeshrai) [inline src/ResourceManager/SiteRecovery/Commands.SiteRecovery.Test/ScenarioTests/SiteRecoveryTests.ps1]: updated the PR.

---
In reply to: [78818997](https://github.com/Azure/azure-powershell/pull/2912#discussion_r78818997) [](ancestors = 78818997,78799081)
